WooCommerce/Classes/ViewRelated/Products/Media/WordPressMediaLibraryPickerDataSource.swift

```diff
@@ -0,0 +1,189 @@
+        // The group never changes
+        return NSNull()
+    }
+
+    func unregisterChangeObserver(_ blockKey: NSObjectProtocol) {
+        onDataChange = nil
+    }
+
+    func unregisterGroupChangeObserver(_ blockKey: NSObjectProtocol) {
+        // The group never changes
+    }
+
+    func loadData(with options: WPMediaLoadOptions, success successBlock: WPMediaSuccessBlock?, failure failureBlock: WPMediaFailureBlock? = nil) {
+        syncingCoordinator.resetInternalState()
+        retrieveMedia(pageNumber: syncingCoordinator.pageFirstIndex, pageSize: Constants.numberOfItemsPerPage) { [weak self] (mediaItems, error) in
+            guard error == nil else {
+                failureBlock?(error)
+                return
+            }
+            self?.mediaItems = mediaItems
+            successBlock?()
+        }
+    }
+
+    func mediaTypeFilter() -> WPMediaType {
+        return .image
+    }
+
+    func ascendingOrdering() -> Bool {
+        return true
+    }
+
+    func add(_ image: UIImage, metadata: [AnyHashable: Any]?, completionBlock: WPMediaAddedBlock? = nil) {}
+
+    func addVideo(from url: URL, completionBlock: WPMediaAddedBlock? = nil) {}
+
+    func setMediaTypeFilter(_ filter: WPMediaType) {}
+
+    func setAscendingOrdering(_ ascending: Bool) {}
+
+    func setSelectedGroup(_ group: WPMediaGroup) {}
+}
+
+extension WordPressMediaLibraryPickerDataSource: SyncingCoordinatorDelegate {
+    func sync(pageNumber: Int, pageSize: Int, reason: String?, onCompletion: ((Bool) -> Void)?) {
+        retrieveMedia(pageNumber: pageNumber, pageSize: pageSize) { [weak self] (mediaItems, error) in
+            guard error == nil else {
+                return
+            }
+            self?.updateMediaItems(mediaItems, pageNumber: pageNumber, pageSize: pageSize)
+        }
+    }
+}
+
+private extension WordPressMediaLibraryPickerDataSource {
+    func retrieveMedia(pageNumber: Int, pageSize: Int, completion: @escaping (_ mediaItems: [Media], _ error: Error?) -> Void) {
+        let action = MediaAction.retrieveMediaLibrary(siteID: siteID,
+                                                      pageNumber: pageNumber,
+                                                      pageSize: pageSize) { (mediaItems, error) in
+                                                        guard mediaItems.isEmpty == false else {
+                                                            completion([], error)
+                                                            return
+                                                        }
+                                                        completion(mediaItems, nil)
+        }
+        ServiceLocator.stores.dispatch(action)
+    }
+
+    /// Appends the new media items to the existing media items.
+    ///
+    func updateMediaItems(_ newMediaItems: [Media], pageNumber: Int, pageSize: Int) {
```

Comment:
I wonder if we should also add some unit tests for the paging in `WordPressMediaLibraryPickerDataSource`. Not sure if that's easily achievable though. This can be a debt issue.

Reply:
yeaa great catch on the missing unit tests! I created a separate issue to add unit tests for `WordPressMediaLibraryPickerDataSource` since this PR is big enough: https://github.com/woocommerce/woocommerce-ios/issues/2074